### PR TITLE
Z-Pattern and Media Blocks

### PR DESCRIPTION
**Pull request by ryanmparrish**
## Description
This covers the HLX implementation of the consonant media and z-pattern blocks. 

We may need to sync up on a few things regarding consonant block shared functions ex.(/decorate/) and any global reaching styles/scripts that may not be needed due to milo and or introducing spectrum global vars etc. LMK if you wanna jam on this together. Happy to refactor anything needed to support cross pollination on these efforts. 100% test coverage. 

I've followed in related subtasks to get a stamp of approval on design. LMK how I we can get better momentum/sign off on the other access/SEO subtasks etc.  Also Im currently seeing a `92` LH score... we might want to powwow on this and dissect what is older consonant repo vs already covered in milo and if theres any issues being introduced. 

### Z-Pattern
Can make small, medium, large variants
Supports detail, headline, image, actions, content
Supports header, intro body copy and alternating media blocks
Supports single or multiple actions
Meets all internal accessibility requirements
Meets all internal SEO requirements
Meets all analytics requirements

Branch URL: https://z-pattern--consonant--adobecom.hlx3.page/docs/blocks/z-pattern

Word Document (Source): https://adobe.sharepoint.com/:w:/r/sites/adobecom/_layouts/15/Doc.aspx?sourcedoc=%7B1449EE24-E8BC-4C15-B255-47B9BF66DC36%7D&file=z-pattern.docx&action=default&mobileredirect=true

### Media

Can make small, medium and large variants
Can make multi-section with 2-up, 3-up and 4-up variants
Supports detail, header, body copy
Supports single or double actions and text link
Supports images or full-width background color/image
Meets all internal accessibility requirements
Meets all internal SEO requirements
Meets all analytics requirements

Branch URL: https://z-pattern--consonant--adobecom.hlx3.page/docs/blocks/media

Word Document (Source): https://adobe.sharepoint.com/:w:/r/sites/adobecom/_layouts/15/Doc.aspx?sourcedoc=%7BCFBA2ADF-AF89-4B56-B182-DCCE61038BF9%7D&file=media.docx&action=default&mobileredirect=true

## Jira tasks
Z-Pattern - https://jira.corp.adobe.com/browse/MWPW-104749
Media -https://jira.corp.adobe.com/browse/MWPW-105636

## Screenshots
Screen Shot 2022-05-23 at 2 09 51 PM

### CODE REVIEW

File: blocks/media/media.css [hunk 1]
```diff
@@ -0,0 +1,287 @@
+
+:root {
+  --padding-s: 40px;
+  --padding-m: 48px;
+  --padding-l: 52px;
+  --gap-xs: 24px;
+  --gap-s: 32px;
+  --gap-m: 40px;
+  --gap-l: 56px;
+}
```
Comment (auniverseaway) [inline blocks/media/media.css:10]: Why not use spacing tokens here?

File: blocks/media/media.css [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+    flex-direction: column-reverse;
+}
+
+.media > .foreground .media-row > div {
+    width: 100%;
+}
+
+/*.media > .foreground .media-row > div:first-of-type {*/
+/*    margin-right: var(--spacing-m);*/
+/*}*/
```
Comment (auniverseaway) [inline blocks/media/media.css:64]: I'd remove if you're not using.

File: blocks/media/media.css [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+  .media.two-up.z-pattern .foreground .media-row > div,
+  .media.three-up.z-pattern .foreground .media-row > div,
+  .media.four-up.z-pattern .foreground .media-row > div {
+    flex: 0 0 50%;
+  }
+
+}
+
+
+@media screen and (min-width: 800px) {
```
Comment (auniverseaway) [inline blocks/media/media.css:259]: This is a non-standard breakpoint...

0-599
600-899
etc.

You can see them in dexter variables.

File: blocks/media/media.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/*
+ * Media - v0.0.1
```
Comment (auniverseaway) [inline blocks/media/media.js:14]: I'd probably sync this up with the actual Consonant version number you're targeting.

**Comment by auniverseaway** [comment]
@ryanmparrish is this all ready? Can we get these merged?

**Review by Sartxi — APPROVED** [review]

**Pull request merged**